server/update-user.js

```diff
@@ -49,7 +49,7 @@ export default function updateUserFactory(analyticsClient) {
       anonymousId = _.first(user.anonymous_ids);
     }
 
-    const publicIdField = public_id_field === "id" ? "id" : "external_id";
+    const publicIdField = public_id_field; // === "id" ? "id" : "external_id";
```

Comment:
can we safely remove that commented part? I assume it was done for cases where we don't have any value in `public_id_field` connector setting. What will happen if we push that change to a connector which is missing the setting?

We may not set the `publicIdField` default value here only if  `ignore_segment_userId` is true, what do you think? I know it's more complicated code but it's safer I think.